### PR TITLE
TC-2558 |  Improved error handling for getBookingByTicketNumber

### PR DESCRIPTION
**Pull request by InkFaust**

### CODE REVIEW

File: src/Services/Air/AirErrors.js [hunk 1]
```diff
@@ -107,8 +107,8 @@ Object.assign(AirRuntimeError, createErrorsList({
   InvalidRequestData: 'Request data is invalid',
   NoValidFare: 'No valid fare for input criteria.',
   TravelersListError: 'Not all BookingTravelers present in list or wrong lookup keys provided',
-  PnrParseError: 'Failed to parse PNR from ticket information request response',
-  GetPnrError: 'Failed to obtain PNR from ticket information',
+  ParseTicketPNRError: ['Failed to parse PNR from ticket information request response', errorCodes.Validation],
+  GetBookingByTicketNumberError: 'Unexpectedly failed to obtain PNR from ticket information',
```
Comment (dchertousov) [inline src/Services/Air/AirErrors.js:111]: Just `Failed to obtain PNR from ticket information` for the same reason.

And I'm not sure about this error. Do we need it? If request to the uAPI failed, why wrap this with another error?

**Review by dchertousov — CHANGES_REQUESTED** [review]
See comments in the PR

File: src/Services/Air/Air.js [hunk 2]
```diff
@@ -287,17 +287,24 @@ module.exports = (settings) => {
...
-        .then(
-          response => terminal.closeSession()
-            .then(() => response.match(/RLOC [^\s]{2} ([^\s]{6})/)[1])
-            .catch(() => Promise.reject(new AirRuntimeError.PnrParseError(response)))
-        )
-        .catch(
-          err => Promise.reject(new AirRuntimeError.GetPnrError(options, err))
-        );
+    async getBookingByTicketNumber(options) {
+      return new Promise(async (resolve, reject) => {
```
[the remark this answers is not in the document]
Comment (InkFaust) [reply src/Services/Air/Air.js:291]: Disagree. Then the first error will be rethrown as another error or I'll need to check the error type to throw one or another, I don't like either of those. I probably don't need `async` in the function name, forgot to remove it, but having promise is better than handling rethrowing IMO.

File: src/Services/Air/Air.js [hunk 3]
```diff
@@ -287,17 +287,24 @@ module.exports = (settings) => {
...
+        try {
+          const terminal = createTerminalService(settings);
+          const screen = await terminal.executeCommand(`*TE/${options.ticketNumber}`) || '';
```
[the remark this answers is not in the document]
Comment (InkFaust) [reply src/Services/Air/Air.js:294]: No mater what `screeen` will be a string, in that case .match never fails.
If we cannot retrieve PNR from the screen, we don't care whatever the screen was a null/undefined/whatever, we only care if we can or cannot retrieve the PNR number. For all other errors thrown during the execution and are unhandled, we use `catch` and reject to eliminate rethrow.

**Review by dchertousov — APPROVED** [review]

**Pull request merged**